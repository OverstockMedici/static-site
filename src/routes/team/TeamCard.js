import React from 'react';

const TeamCard = ({ name, image, title, info }) => {
    return (
        <figure data-testid="team-card" key={name}>
            <img src={image} alt={name} />
            <figcaption>
                <div className="employee-name">{name}</div>
                <div className="employee-title">/ {title}</div>
                <div className="employee-info">{info}</div>
            </figcaption>
        </figure>
    );
};

export default TeamCard;
