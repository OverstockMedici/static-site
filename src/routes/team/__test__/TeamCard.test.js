import React from 'react';
import ReactDOM from 'react-dom';
import TeamCard from './../TeamCard';

import { render, cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const figure = document.createElement('figure');
    ReactDOM.render(<TeamCard />, figure);
});

it('renders TeamCard correctly', () => {
    const { getByTestId, container } = render(
        <TeamCard
            name="Jonathan Johnson"
            title="Cool Guy"
            info="Does cool things"
        />,
    );
    expect(getByTestId('team-card')).toHaveTextContent('Jonathan Johnson');
    expect(getByTestId('team-card')).toHaveTextContent('Cool Guy');
    expect(getByTestId('team-card')).toHaveTextContent('Does cool things');
});

it('matches snapshot', () => {
    const teamCardTree = renderer
        .create(
            <TeamCard
                name="Jonathan Johnson"
                title="Cool Guy"
                info="Does cool things"
            />,
        )
        .toJSON();

    expect(teamCardTree).toMatchSnapshot();
});
