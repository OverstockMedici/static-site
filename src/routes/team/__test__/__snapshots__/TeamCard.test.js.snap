// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<figure
  data-testid="team-card"
>
  <img
    alt="Jonathan Johnson"
  />
  <figcaption>
    <div
      className="employee-name"
    >
      Jonathan Johnson
    </div>
    <div
      className="employee-title"
    >
      / 
      Cool Guy
    </div>
    <div
      className="employee-info"
    >
      Does cool things
    </div>
  </figcaption>
</figure>
`;
